sklearn/metrics/pairwise.py

```diff
@@ -765,10 +765,12 @@ def manhattan_distances(X, Y=None, sum_over_features=True):
 
         X = csr_matrix(X, copy=False)
         Y = csr_matrix(Y, copy=False)
+        X.sort_indices()
```

Comment:
formally you might require sum_duplicates too??? Perhaps we should make a copy if the matrix indices are unsorted, rather than modifying inplace without permission..

Reply:
While I agree that doing inplace modification of input is bad, I can't think of a case where it would be bad to sort_indices and sum_duplicates for a CSR array. scipy is somewhat ambiguous about doing that at initialization https://github.com/scipy/scipy/blob/26b7d3f40905d85845a3af75a67c318a8d441ed1/scipy/sparse/compressed.py#L198

Maybe adding a note to the docstring about it could be enough? Of course if the arrays are read-only a copy still need to be triggered.

Reply:
So let's sum duplicates here and document the in place operation